pkg/backporter/backporter.go

```diff
@@ -553,30 +574,111 @@ func CreateCloneHandler(client bugzilla.Client, allTargetVersions sets.String, m
 			handleError(w, err, fmt.Sprintf("unable to fetch bug details- Bug#%d", bugID), http.StatusNotFound, endpoint, bugID, m)
 			return
 		}
-		// Create a clone of the bug
-		cloneID, err := client.CloneBug(bug)
+		absentReleaseErrMsg := fmt.Sprintf("invalid argument - %s is not a valid TargetRelease, must be one of %v", req.FormValue("release"), allTargetVersions.List())
+		if !allTargetVersions.Has(req.FormValue("release")) {
+			handleError(w, fmt.Errorf(absentReleaseErrMsg), absentReleaseErrMsg, http.StatusBadRequest, endpoint, bugID, m)
+			return
+		}
+
+		// Get clones and sort them
+		clones, err := client.GetAllClones(bug)
+		if err != nil {
+			handleError(w, err, fmt.Sprintf("unable to convert \"ID\" parameter from string to int: %s", req.FormValue("ID")), http.StatusBadRequest, req.URL.Path, 0, m)
+			return
+		}
+		sortByTargetRelease(clones)
+		releaseInvalidErrorMsg := "%s cannot be parsed as release, must be of the form X.X.X"
+		toCloneMajorMinorRelease, err := getMajorMinorRelease(req.FormValue("release"))
 		if err != nil {
-			handleError(w, err, "clone creation failed", http.StatusInternalServerError, endpoint, bugID, m)
+			handleError(w, err, fmt.Sprintf(releaseInvalidErrorMsg, req.FormValue("release")), http.StatusBadRequest, endpoint, bugID, m)
 			return
 		}
-		targetRelease := bugzilla.BugUpdate{
-			TargetRelease: []string{
-				req.FormValue("release"),
-			},
+		var sourceBug *bugzilla.Bug
+		var sourceBugMajorMinorRel string
+		for _, clone := range clones {
+			if !isTargetReleaseSet(clone) {
+				continue
+			}
+			cloneMajorMinorRel, err := getMajorMinorRelease(clone.TargetRelease[0])
+			if err != nil {
+				handleError(w, err, fmt.Sprintf(releaseInvalidErrorMsg, clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if toCloneMajorMinorRelease == cloneMajorMinorRel {
+				handleError(w, err, fmt.Sprintf("clone for major release %s already exists", clone.TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
+				return
+			}
+			if cloneMajorMinorRel > toCloneMajorMinorRelease {
+				sourceBug = clone
+				sourceBugMajorMinorRel = cloneMajorMinorRel
+				break
+			}
 		}
-		// Updating the cloned bug with the right target version
-		if err := client.UpdateBug(cloneID, targetRelease); err != nil {
-			handleError(w, err, fmt.Sprintf("clone created - Bug#%d, but failed to specify version for the cloned bug.", cloneID), http.StatusInternalServerError, endpoint, bugID, m)
+		if sourceBug == nil {
+			handleError(w, err, fmt.Sprintf("one bug with greater release needs to be present to clone from (highest release present is %s)", clones[len(clones)-1].TargetRelease[0]), http.StatusBadRequest, endpoint, bugID, m)
 			return
 		}
+		var sortedMajorRelease []string
+		for release := range allTargetVersions {
```

Comment:
allTargetVersions is a set and using the set makes sense for when I want to lookup stuff/ remove stuff from it

Reply:
Yeah I agree but its a static set so constructing the sorted list once rather than during each request would make more sense or not?

Reply:
There are two things being done here
1. Its used to populate the dropdown where we would need a set
2. Its used for determining the things which would need to be cloned for which a static sorted list would be fine
I could  pass in two things(a sorted list and a set) for that, or I could construct the set from the sorted list I pass in.
I'll do option 2 unless you say otherwise

Reply:
Its a tiny amount of data and its completely static during runtime, just store both the set and the sorted list, no reason to construct one of them in the handler

Reply:
I would need a separate copy of the set anyway since I am modifying it. So I would have to construct a set anyway - the only difference would be whether its from a list or a set. Thats why passing in the sorted list as you suggested would be enough